app.ts

```
@@ -0,0 +1,326 @@
+			throw error;
+		});
+}
+
+// Fetch a range of records from the server based on specified indices.
+function fetchRecords(fromRecord: number, toRecord: number): Promise<any> {
+	if (fromRecord > toRecord) {
+		fromRecord = toRecord
+	}
+	return fetch(`${state.IMQS}/records?from=${fromRecord}&to=${toRecord}`)
+		.then((response) => response.json())
+		.catch((error) => {
+			throw error;
+		});
+}
+
+// Calculate the number of records that can fit on the screen.
+function screenCalculations(screenHeight: number): number {
+	const estimatedRowHeightFactor = 1;
+	const estimatedRowHeight = estimatedRowHeightFactor * 50;
+	const recordsPerPage = Math.floor(screenHeight / estimatedRowHeight) - 1;
+	return Math.max(recordsPerPage - 2, 1);
+}
+
+// Handles the input for the search
+function inputHandling(): Promise<void> {
+	return totalRecords()
+		.then((totalRecCount) => {
+			const searchInput = document.getElementById("searchInput") as HTMLInputElement;
+			searchInput.max = (totalRecCount - 1).toString();
+		});
+}
+
+function debounce(func: any, delay: number) {
+	let timeoutId: any;
+	return function (...args: any) {
+		clearTimeout(timeoutId);
+		timeoutId = setTimeout(() => {
+			func(...args);
+		}, delay);
+	};
+}
+
+// Display records on the page based on the specified range.
+async function displayData(fromRecord: number, recordsDisplayed: number) {
+	$("#loader").show();
+	$("#tableWrapper").hide();
+	let isSearchMode = state.searchedIndex !== null;
+	const searchIndexOnPage = isSearchMode ? state.searchedIndex! - fromRecord : 0;
+	fromRecord = Math.max(fromRecord, 0);
+	const recordCount = await totalRecords();
+	const maximumRecords = Math.min(recordCount - fromRecord, recordsDisplayed);
+	const toRecord = fromRecord + maximumRecords - 1;
+	const data = await fetchRecords(fromRecord, toRecord);
+	let tableData = "";
+	if (isSearchMode) {
+		if (currentPage * state.recordsPerPage > recordCount) {
+			recordsDisplayed = recordCount - fromRecord;
```

I don't agree with changing the amount of records that need to be displayed. The only scenario where something like that would be ok, is if the the amount of records asked to be displayed is larger than the total records in the back-end.

But in that case I would rather throw an error.